autogen/agentchat/conversable_agent.py

```diff
@@ -778,6 +778,11 @@ def clear_history(self, recipient: Optional[Agent] = None, nr_messages_to_preser
         if recipient is None:
             if nr_messages_to_preserve:
                 for key in self._oai_messages:
+                    # if breaking history between function call and function response, save function call message
+                    # additionally, otherwise openai will return error
+                    first_msg_to_save = self._oai_messages[key][-nr_messages_to_preserve]
```

Comment:
Should we also check for this when `recipient` is not `None`?

Reply:
@ekzhu Well, it's a good question. `recipient` Is never provided when we calling 'clear history' from command prompt, during chat operation. `clear_history` with `recepient` provided is called in very few cases, before the chat instantiation I think (I don't really understand why that needed, maybe someone else could provide better explanation). So basically `clear_history(recepient)` will be never called from tool response. If I wrong, please correct me. Anyway, If needed, I can add that check where `recipient` present, it's not much work.